### PR TITLE
feat(sonarqube): RHICOMPL-1260 add initial sonar config

### PR DESCRIPTION
**Pull request by akofink**
Signed-off-by: Andrew Kofink <akofink@redhat.com>

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#sonar.projectVersion=1.0
+ 
+# Path is relative to the sonar-project.properties file. Defaults to .
+#sonar.sources=.
+ 
+# Encoding of the source code. Default is default system encoding
+#sonar.sourceEncoding=UTF-8
+
+sonar.javascript.lcov.reportPaths=coverage/lcov.info
+sonar.coverage.exclusions=config/*,src/__fixtures__/*,src/**/**/*.test.js,src/App*.js,*.config.js,src/bootstrap*.js,src/entry*.js,src/Routes.js
```
Comment (akofink) [inline sonar-project.properties:18]: @bastilian anything here that shouldn't be? None of these were covered, so I suspected they're not expected to be tested.

Comment (Victoremepunto) [reply sonar-project.properties:18]: We already have our test coverage reports in Codecov, do we still wanna do this, or we're going to migrate this stats to Sonarqube ? it doesn't makes much sense to have this stats reported in several reporting tools, however I don't have strong opinions about it

Comment (akofink) [reply sonar-project.properties:18]: There is definitely duplication here, but sonarqube uses the output of our tests and coverage to report its own coverage - it doesn't run the tests itself or measure coverage itself.

Comment (bastilian) [reply sonar-project.properties:18]: @akofink Is there maybe a way to configure it the other way around and configure what to include, not exclude, to match (the not up to date yet) patterns jest uses? https://github.com/RedHatInsights/compliance-frontend/blob/master/package.json#L51

**Review by vkrizan — APPROVED** [review]
Thank you!

File: sonar-project.properties (hunk 1 above)
Comment (akofink) [reply sonar-project.properties:18]: Unfortunately not. There is only an exclusion list. I'll update these exclusions to match the package.json.

Comment (akofink) [reply sonar-project.properties:18]: Hmm.. that exclusions list does not seem to be quite correct. For example, do you want to track coverage on *.test.js files?

Comment (akofink) [reply sonar-project.properties:18]: We can always update this list in the future if we have improvements.

**Pull request merged**